src/internal/connector/graph/concurrency_limiter.go

```diff
@@ -0,0 +1,52 @@
+package graph
+
+import (
+	"net/http"
+	"sync"
+
+	khttp "github.com/microsoft/kiota-http-go"
+)
+
+// ConcurrencyLimiter middleware limits the number of concurrent requests to graph API.
+// Currently we only enforce concurrency limit for exchange
+type ConcurrencyLimiter struct {
+	semaphoreExch chan struct{}
+}
+
+var (
+	once               sync.Once
+	concurrencyLimiter *ConcurrencyLimiter
+	// Outlooks expects max 4 concurrent requests
+	// https://learn.microsoft.com/en-us/graph/throttling-limits#outlook-service-limits
+	maxConcurrentRequestsExchange = 4
+)
+
+func InitializeConcurrencyLimiter(capacity int) {
+	once.Do(func() {
+		if capacity < 1 || capacity > maxConcurrentRequestsExchange {
+			capacity = maxConcurrentRequestsExchange
+		}
+
+		concurrencyLimiter = &ConcurrencyLimiter{
+			semaphoreExch: make(chan struct{}, capacity),
+		}
+	})
+}
+
+func GetConcurrencyLimiter() *ConcurrencyLimiter {
```

Comment:
Seems like an unnecessary func, unless we expect external packages to call it?  Again, exporting these values feels like a smell.

Reply:
Good catch! Thank you!